src/modules/Elsa.Workflows.Runtime/Requests/DispatchWorkflowDefinitionRequest.cs

```diff
@@ -60,5 +60,5 @@ public DispatchWorkflowDefinitionRequest(string definitionId, VersionOptions ver
     /// <summary>
     /// The ID of the activity that triggered the workflow.
     /// </summary>
-    public string? TriggerActivityId { get; init; }
+    public string? TriggerActivityId { get; set; }
```

Comment:
Why should we want to change this after initialization?

Reply:
The rationale here is that this type has default constructor and a constructor that accepts the two required members.

This allows client code to instantiate a request object and provide the two required values via the constructor, without being forced to initialize the remaining properties, which are optional, via the initializer. For example, it allows for this usage pattern:

```csharp
var request = new DispatchWorkflowDefinitionRequest("my definition ID", VersionOptions.FromVersion(1));

// Configure other props.
request.TriggerActivityId = "activity-1";
```

In contrast, if the optional properties must be set during object init, the code would **have** to look like this:

```csharp
var request = new DispatchWorkflowDefinitionRequest("my definition ID", VersionOptions.FromVersion(1))
{
   TriggerActivityId = "activity-1"
}
```

Or this:

```csharp
var request = new DispatchWorkflowDefinitionRequest
{
   DefinitionId = "my definition ID",
   VersionOptions.FromVersion(1),
   TriggerActivityId = "activity-1"
}
```

All these options are fine, but subject to user preference.

Reply:
I would prefer it to still have it on init so the value can not be changed after initialization. This id will not change after the request object has been made. I those cases I prefer to make the properties immutable, regardless if they are optional.